source-reddit-fetcher/source_reddit_fetcher/source.py

```diff
@@ -0,0 +1,247 @@
+        super().__init__(token) 
+
+
+
+class ApiStream(HttpStream):
+
+    primary_key: Optional[str] = None
+    url_base = "https://oauth.reddit.com/"
+
+    def __init__(self, authenticator: requests.auth.AuthBase):
+        super().__init__(authenticator=authenticator)
+
+
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        
+        wait = response.headers.get("Retry-After")
+        if not wait:
+            wait = response.headers.get("x-ratelimit-reset")
+        
+        if response.status_code == 429:
+            logger.warning(f"Raised too many requests at once! Waiting {wait}s...")
+        
+        return float(wait) if wait else None
+
+
+    def to_utc_timestamp(self, timestamp: float) -> datetime.date:
+        return datetime.datetime.fromtimestamp(timestamp, tz=datetime.timezone.utc).date()
+
+
+class SubredditPosts(ApiStream):
+
+    # Based on "type prefixes"
+    # https://www.reddit.com/dev/api/
+    type_prefixes = {
+        "t1": "comment",
+        "t2": "account",
+        "t3": "link",
+        "t4": "message",
+        "t5": "subreddit",
+        "t6": "award"
+    }
+
+    primary_key = "id"
+    cursor_field = "created_timestamp"
+
+    def __init__(self, subreddit: str, authenticator: requests.auth.AuthBase):
+        super().__init__(authenticator)
+        self.subreddit = subreddit
+
+
+
+    def path(self, **kwargs):
+        return f"{self.url_base}/r/{self.subreddit}/new"
+
+
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        
+        data: dict[str, Any] = response.json()
+        children: list[dict] = data.get("data", {}).get("children", [])
+
+        for child in children:
+            
+            data = child.get("data", {})
+            if not data:
+                continue
+            
+            row = {
+                "id": f"{self.subreddit}-{data['id']}",
+                "kind_tag": child["kind"],
+                "kind_name": self.type_prefixes[child["kind"]],
+                "subreddit": self.subreddit,
+                "post_id": data["id"],
+                "post_url": BASE_URL + data["permalink"],
+                "url": data["url"],
+                "domain": data["domain"],
+                "created_timestamp": datetime.datetime.fromtimestamp(data["created_utc"], tz=datetime.timezone.utc),
+                "timezone": "UTC",
+                "title": data["title"],
+                "text": data["selftext"],
+                "html_text": data["selftext_html"],
+                "author": data["author"],
+                "raw": json.dumps(data)
+            }
+            yield row
+
+
+
+    def next_page_token(self, response: requests.Response) -> Optional[dict[str, Any]]:
+        data: dict = response.json()
+        after = data.get("data", {}).get("after")
+        return {"after": after} if after else None
+
+
+
+class SubredditVotes(SubredditPosts):
```

Comment:
```suggestion
class Votes(HttpSubStream, Posts):
```
If you want it to depend of a instance of post